diff --git a/node_modules/react-native-windows/Microsoft.ReactNative/Modules/NativeUIManager.cpp b/node_modules/react-native-windows/Microsoft.ReactNative/Modules/NativeUIManager.cpp
index fb25176..eb1201a 100644
--- a/node_modules/react-native-windows/Microsoft.ReactNative/Modules/NativeUIManager.cpp
+++ b/node_modules/react-native-windows/Microsoft.ReactNative/Modules/NativeUIManager.cpp
@@ -309,16 +309,10 @@ static YGValue YGValueOrDefault(
       return YGValue{static_cast<float>(pct.asDouble()), YGUnitPercent};
     }
     if (str.length() > 2 && (str.compare(str.length() - 2, 2, "pt") || str.compare(str.length() - 2, 2, "px"))) {
-      shadowNode.RedBox(
-          "Value '" + value.AsString() + "' for " + key + " is invalid. Cannot be converted to YGValue. '" +
-          str.substr((str.length() - 2), 2) + "' unit not needed. Simply use integer value.");
       return defaultValue;
     }
   }
 
-  shadowNode.RedBox(
-      "Value '" + value.AsString() + "' for " + key +
-      " is invalid. Cannot be converted to YGValue. Did you forget the %? Otherwise, simply use integer value.");
   return defaultValue;
 }
 
