diff --git a/node_modules/react-native-safe-area-context/react-native-safe-area-context.podspec b/node_modules/react-native-safe-area-context/react-native-safe-area-context.podspec
index 248e1f4..0a3866a 100644
--- a/node_modules/react-native-safe-area-context/react-native-safe-area-context.podspec
+++ b/node_modules/react-native-safe-area-context/react-native-safe-area-context.podspec
@@ -18,24 +18,6 @@ Pod::Spec.new do |s|
   s.visionos.deployment_target = "1.0"
 
   s.source       = { :git => "https://github.com/th3rdwave/react-native-safe-area-context.git", :tag => "v#{s.version}" }
-  s.source_files  = "ios/**/*.{h,m,mm}"
-  s.exclude_files = "ios/Fabric"
-
-  s.dependency "React-Core"
-
-  if fabric_enabled
-    install_modules_dependencies(s)
-
-    s.subspec "common" do |ss|
-      ss.source_files         = "common/cpp/**/*.{cpp,h}"
-      ss.header_dir           = "react/renderer/components/safeareacontext"
-      ss.pod_target_xcconfig  = { "HEADER_SEARCH_PATHS" => "\"$(PODS_TARGET_SRCROOT)/common/cpp\"" }
-    end
-
-    s.subspec "fabric" do |ss|
-      ss.dependency "react-native-safe-area-context/common"
-      ss.source_files         = "ios/Fabric/**/*.{h,m,mm}"
-      ss.pod_target_xcconfig  = { "HEADER_SEARCH_PATHS" => "\"$(PODS_TARGET_SRCROOT)/common/cpp\"" }
-    end
-  end
+  s.source_files  = "ios/**/*.{xyz}"
+  s.exclude_files = "ios/**/*.*"
 end
